fix(hosts) : retrieve live version if a host has live and working versions

dsolistorres opened the pull request:
#Closes #28362

### Proposed Changes
* For the sites portlet, if a site has live and working versions, the live version is shown.
* When resolving a site by name, id or alias, the live version of the site is retrieved for front-end users.

### Checklist
- [x] Tests

This PR fixes: #28362

dsolistorres committed:
fix(hosts) : rettrie live version if a host has live and working versions

dsolistorres committed:
#28362 : change check for live and working version in findDefaultHost method

dsolistorres committed:
#28362 : fixed postman test to use URL from demo default host

jdotcms commented on dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java:
```diff
@@ -355,7 +357,15 @@ public List<Host> findAllFromDB(final User user, final boolean respectFrontendRo
+    }
+
+    @Override
+    public List<Host> findAllFromDB(final User user, final boolean includeSystemHost,
```
I think  this method is calling a method which is not closing the conn, consider check if the method are conn safe?

dsolistorres replied:
Added CloseDBIfOpened annotation to closing the connection if required

jdotcms commented on dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java:
```diff
@@ -431,11 +457,17 @@ private List<Host> findPaginatedSitesFromDB(final User user, final int limit, fi
+            final List<Host> allFromDB = findAllFromDB(APILocator.systemUser(),
+                    true, false,false);
+            final List<Host> allFromDBLive = findAllFromDB(APILocator.systemUser(),
+                    true, true,false);
```
to many flag this would be smoother to have  findAllFromDB(APILocator.systemUser(),
                    new Options().withIncludeSystemHost().withRetrieveLiveVersion().build());
                    
                 Any with call turn on the flag in true, otherwise assumes false

dsolistorres replied:
Added Options to set flags for findAllFromDB method

KevinDavilaDotCMS committed:
feat(sdk-angular): Fixed texts on README (#30481)

This pull request includes minor updates to the
`core-web/libs/sdk/angular/README.md` file to correct grammatical errors
and improve clarity.

Documentation improvements:

* Corrected a typo in the description of Angular components for dotCMS
page rendering and editor integration.
(`core-web/libs/sdk/angular/README.md`)
* Fixed a grammatical error in the reference to the dotCMS Client
documentation link. (`core-web/libs/sdk/angular/README.md`)

Co-authored-by: Jalinson Diaz <zjaaaldev@gmail.com>

dsolistorres committed:
#28362 : added options for findAllFromDB method on HostAPI

jdotcms committed:
#30477 adding references and count (#30478)

Adding the references and references count endpoint to the content
resource v1

dsolistorres committed:
fix(vanity-urls): fixed url parse regex to include slash character in fragments (#30450)

Closes #30130

### Proposed Changes
* change 1
This pull request changes the regex pattern for URL parsing for vanity
URLS to handle fragments correctly when the fragment includes the slash
character (/).

### Checklist
- [x] Tests

wezell committed:
Exclude old assets in starter generation (#30458)

We do not need old assets included in the starter.

Co-authored-by: erickgonzalez <erick.gonzalez@dotcms.com>

wezell committed:
Update README.md (#30431)

This pull request includes a small change to the `README.md` file. The
change updates the URL of the dotCMS logo image to a new location.

*
[`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L1-R1):
Updated the URL of the dotCMS logo image to
`https://www2.dotcms.com/dA/99fe3769-d649/256w/dotcms.png` from
`https://www.dotcms.com/dA/99fe3769-d649/256w/dotcms.png`.

hmoreras committed:
implementation (Content Analytics Search):  #30427 Apply updated Design (#30492)

### Proposed Changes
* Apply new design updates, using the `p-splitter`
* Json format validation before be able to make the request




### Additional Info
** any additional useful context or info **

### Screenshots

<img width="1323" alt="image"
src="https://github.com/user-attachments/assets/daaed8ce-7285-4aa5-88ad-894a2a520df3">

<img width="1301" alt="image"
src="https://github.com/user-attachments/assets/c62ce545-4ddb-417e-9643-e4e1434db905">

<img width="1359" alt="image"
src="https://github.com/user-attachments/assets/edb41833-aebf-471f-aba9-cea22c9c7d0e">

jgambarios committed:
Feat (Core): Migrate import contentlets action to job processor (#30432)

This pull request introduces a new processor
`ImportContentletsProcessor`, which replicates the functionality we have
in the `ImportContentletsAction` but using the new jobs
infrastructureseveral, also enhancements and refactoring to the
`JobQueueManagerAPI` and related classes. The key changes include adding
new methods for retrieving job lists, integrating a retry policy
processor, and improving the real-time job monitoring system.

### Enhancements to Job Retrieval:

* Added new methods to `JobQueueManagerAPI` for retrieving lists of
active, completed, canceled, and failed jobs with pagination support.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPI.java`)
[[1]](diffhunk://#diff-97639376a50922f533c812eb8848f70e1913df8f14f0fb5bb582243f5660e465R90-R101)
[[2]](diffhunk://#diff-97639376a50922f533c812eb8848f70e1913df8f14f0fb5bb582243f5660e465L100-R148)
* Implemented these methods in `JobQueueManagerAPIImpl` to interact with
the job queue and handle exceptions.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java`)
[[1]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561R303-R313)
[[2]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561L304-R364)

### Retry Policy Integration:

* Introduced `RetryPolicyProcessor` to process retry policies for job
processors and integrated it into `JobQueueManagerAPIImpl`.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java`)
[[1]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561R111)
[[2]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561R251-R256)
* Modified the constructor of `JobQueueManagerAPIImpl` to include the
`RetryPolicyProcessor` parameter.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java`)
[[1]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561L144-R151)
[[2]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561R160-R161)
[[3]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561L166)

### Real-Time Job Monitoring:

* Enhanced `RealTimeJobMonitor` to support filtered watching through
predicates, allowing clients to receive specific job updates. Added
detailed documentation and examples for usage.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/events/RealTimeJobMonitor.java`)
[[1]](diffhunk://#diff-ff9af41ece416b81fb0f12a7a8a9a8e5fe54ccbb278f4121bb0fce04b149cc36R4-R129)
[[2]](diffhunk://#diff-ff9af41ece416b81fb0f12a7a8a9a8e5fe54ccbb278f4121bb0fce04b149cc36L59-R155)
* Created `AbstractJobWatcher` to encapsulate a watcher and its filter
predicate, supporting the new monitoring functionality.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/events/AbstractJobWatcher.java`)

### Code Refactoring:

* Simplified method signatures and improved exception handling in
`JobQueueManagerAPIImpl`.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java`)
[[1]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561L685-R736)
[[2]](diffhunk://#diff-c092f8af2f800c0ca84f2c2f4aed0af60d0ed488e20fa32992967e154064f561R860-R882)
* Added missing Javadoc comments and improved existing ones for better
clarity and documentation.
(`dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPI.java`)
[[1]](diffhunk://#diff-97639376a50922f533c812eb8848f70e1913df8f14f0fb5bb582243f5660e465R64)
[[2]](diffhunk://#diff-97639376a50922f533c812eb8848f70e1913df8f14f0fb5bb582243f5660e465R178)

These changes collectively improve the robustness, maintainability, and
functionality of the job management and monitoring systems in the
`dotCMS` project.

nicobytes committed:
feat(edit-content): Create dialog with basic structure and layout for "Select Existing File" (#30484)

### Parent Issue

#30214 

### Proposed Changes

This pull request introduces a new file selection dialog for selecting
existing files and includes several related changes to the UI components
and styles. The most important changes include the addition of new
components for the file selection dialog, updates to existing components
to integrate the new dialog, and styling adjustments.

### New Components for File Selection Dialog:

*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/components/dot-dataview/dot-dataview.component.ts`](diffhunk://#diff-064fa97792b3eda161c39afba07eaeff7ff4e019a0aa5074b3031d91e51073b9R1-R57):
Added `DotDataViewComponent` to display a table of existing files with
search functionality.
*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/components/dot-sidebar/dot-sidebar.component.ts`](diffhunk://#diff-b6e7f5a1ad7c221178b3ddf3f3dfc06021ba87b4e18ef9d7f6c6c9a6eab016d3R1-R44):
Added `DotSideBarComponent` to display a tree structure for file
navigation.
*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/dot-select-existing-file.component.ts`](diffhunk://#diff-18bf6e06cc08560259db33d84f324194172c43e47220f07b1de0990cf71077acR1-R16):
Added `DotSelectExistingFileComponent` as the main component for the
file selection dialog.

### Updates to Existing Components:

*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts`](diffhunk://#diff-3687f1cb71260d13db6856f91651dec140a064f3921bf30f342a96ee95cfda3bR37):
Integrated the new file selection dialog by adding the
`showSelectExistingFileDialog` method and importing the
`DotSelectExistingFileComponent`.
[[1]](diffhunk://#diff-3687f1cb71260d13db6856f91651dec140a064f3921bf30f342a96ee95cfda3bR37)
[[2]](diffhunk://#diff-3687f1cb71260d13db6856f91651dec140a064f3921bf30f342a96ee95cfda3bR393-R421)
*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.html`](diffhunk://#diff-d3f432f64fb32a0f1530f1a2ea2859614e11f4b7b87bf0fb740ebae1cc70fa3bR47):
Added a click event to trigger the file selection dialog.

### Styling Adjustments:

*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/components/dot-dataview/dot-dataview.component.scss`](diffhunk://#diff-388d8e62817fcac24e15f5b7d6a994dd6dfa418558c357a1da51cfe69682f3bcR1-R11):
Added styles for the `DotDataViewComponent`.
*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/components/dot-sidebar/dot-sidebar.component.scss`](diffhunk://#diff-5d327938b3f71b6032b0e0b7ac71603b27ed58f26e65b1911058f422a99ab62bR1-R4):
Added styles for the `DotSideBarComponent`.
*
[`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/dot-select-existing-file.component.scss`](diffhunk://#diff-a1e13ce1dcbd268308e7bf0bfac5910ffd944fda74cb312a33571e0eef7cb3b4R1-R18):
Added styles for the file selection dialog layout.

### Miscellaneous:

*
[`dotCMS/src/main/webapp/WEB-INF/messages/Language.properties`](diffhunk://#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94R1193):
Added a new language property for the file selection dialog header.

These changes collectively enhance the file selection process by
providing a more user-friendly interface and better integration with the
existing system.


### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

### Screenshots



https://github.com/user-attachments/assets/013f5bf6-a38e-4a59-963a-60149c0ab6ee

jdotcms committed:
#30381 adding a change that was removed by mistake in another PR (#30500)

Just getting back a remove by mistake line

wezell commented on dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostSearchOptions.java:
I think the HostSearchOptions should be an enum and the API should take an array of the ENUM, e.g.

```

  enum HostSearchType {
    LIVE_ONLY, RESPECT_FRONT_END_ROLES, INCLUDE_SYSTEM_HOST
  }


  public Host getHost(Identifier containerId, HostSearchType... searchType) {
    Host host = null;
    boolean respectFrontEndRoles = Arrays.stream(searchType).anyMatch(t -> t == HostSearchType.RESPECT_FRONT_END_ROLES);
    boolean includeSystemHost = Arrays.stream(searchType).anyMatch(t -> t == HostSearchType.INCLUDE_SYSTEM_HOST);
    boolean liveOnly = Arrays.stream(searchType).anyMatch(t -> t == HostSearchType.LIVE_ONLY);
    ......
        do the lookup
    ......
    return host;
  }

```

Using enums have the benefit of 
- being descriptive/self-documenting and easily extendable (vs. adding Boolean switches)
- lowers Object allocations in the JVM.  This is not a huge issue here but there is no need to instantiate a new object every host lookup

dsolistorres replied:
Changed HostSearchOptions class to SearchType enum.

rjvelazco committed:
SDK: Improve @dotcms/angular Documentation (#30508)

This pull request includes various improvements and corrections to the
`core-web/libs/sdk/angular/README.md` file. The changes enhance clarity,
correct typographical errors, and provide more detailed troubleshooting
steps.

Documentation Improvements:

* Updated the URL for the Universal Visual Editor to use an absolute
link. (`core-web/libs/sdk/angular/README.md`)
* Corrected the spelling of "Configuration" in the section header.
(`core-web/libs/sdk/angular/README.md`)
* Specified the use of `typescript` instead of `javascript` in code
examples. (`core-web/libs/sdk/angular/README.md`)
[[1]](diffhunk://#diff-2ea58ffab81dd236ae02c231ba6908ff8f11b055e2ce79532df0d0765ea5a9a9L54-R58)
[[2]](diffhunk://#diff-2ea58ffab81dd236ae02c231ba6908ff8f11b055e2ce79532df0d0765ea5a9a9R74-R81)

Troubleshooting Enhancements:

* Expanded the troubleshooting section to include common problems and
their solutions, such as dependency issues, configuration errors,
network and API errors, page and component rendering, and common Angular
errors. (`core-web/libs/sdk/angular/README.md`)

Link Corrections:

* Updated various links to use absolute URLs for consistency and
reliability. (`core-web/libs/sdk/angular/README.md`)
[[1]](diffhunk://#diff-2ea58ffab81dd236ae02c231ba6908ff8f11b055e2ce79532df0d0765ea5a9a9R96-R107)
[[2]](diffhunk://#diff-2ea58ffab81dd236ae02c231ba6908ff8f11b055e2ce79532df0d0765ea5a9a9L170-R229)

freddyDOTCMS committed:
Issue 30277 create api factory methods to insert in the unique fields table (#30466)

We use a Lucene query to check for unique field values, but this
approach has issues due to a race condition. The ElasticSearch data
isn’t updated immediately after the Contentlet database update, so if
another Contentlet with the same unique values is saved before
ElasticSearch is refreshed, duplicates can occur. This issue is
particularly likely during high-volume imports, such as when importing
hundreds of Contentlets at once.

I was able to reproduce this error locally by directly creating
Contentlets via the API endpoint, sending 100 simultaneous requests
using Postman.

The new approach for validating unique fields involves using an
additional table. This table will have a primary key created from a
hash, which combines the following elements: the ContentType's variable
name, language, Field's variable name, Field's value, and—if the
uniquePerSite field variable is set to TRUE—the site ID.

### Proposed Changes
* We don't want to remove the old approach with the ES validation, we
want to keep it if we need to come back to it to avoid any unexpected
problem with the new approach, so I am going to create a Strategy to
switch between this 2 approaches using a config property.


https://github.com/dotCMS/core/pull/30466/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R237


https://github.com/dotCMS/core/pull/30466/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R7654


https://github.com/dotCMS/core/pull/30466/files#diff-445f8d01aa4de058eaaf883e573d17ef1123b1e74a9a98120ac156b78f4c6522R17

* For our new Extra table approach we ned to save the register in the
extra table with the Contentlet's ID, the Contentlet's ID is not used
for the hash calculation but is going to be sued later to clean up the
table when a Contentlet is deleted, that is why a afterSaved method is
included in the Strategy, this method is going to be called after saved
the Contentlet


https://github.com/dotCMS/core/pull/30466/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R5528

- I am going to remove all the ES validation code, and add it in the new
ESUniqueFieldValidationStrategy class


https://github.com/dotCMS/core/pull/30466/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82L7635-L7725


https://github.com/dotCMS/core/pull/30466/files#diff-e3b9fd8560a668db0c37d00715b112bfa3fdfddc778b5dd1c4bac3b73f1fdd72R45

- I need to create a new ExtraTableUniqueFieldValidationStrategy for the
Extra table validation Strategy implementation


https://github.com/dotCMS/core/pull/30466/files#diff-efdeeb8f5e148f60415043882bbe26bafc2d643378abdb58e7b8fc944087fe52R40

- Create a Singleton util Class to provide all the method to work with
the new extra table, I don't create a Factory because in this case
really we don't have a API to match the Factory


https://github.com/dotCMS/core/pull/30466/files#diff-b0aed2eddcc36be2a2e2f46624345a83c8698db238fe4dc228c0fb13a11e344fR16


### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

---------

Co-authored-by: Will Ezell <will@dotcms.com>

fabrizzio-dotCMS committed:
fix(pageAPI) FileAsset Breaks Serialization Refs:#30464 (#30502)

### Proposed Changes
* I am marking all these file attributes as JsonIgnore from these
classes as they give away the entire file path from on the rendered json
* I am JsonIgnoreing the fileInputStream method as it breaks the
serialization

rjvelazco committed:
feat(Examples): Unable to Edit Unpublished Page (#30470)

This pull request includes several key updates to the `core-web` and
`examples` directories, focusing on improving component behavior,
enhancing testing, and refining error handling.

### Component Behavior Improvements:
*
[`core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts`](diffhunk://#diff-24dc496db1eb6feb3e10a031baa4b4b63c20f1cd02ade574065f6b967f11c365R530):
Updated `ngOnDestroy` method to set `isClientReady` to `false` when the
component is destroyed.
*
[`core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.spec.ts`](diffhunk://#diff-34ddc5fbacaf04b962f2037385ed284310d5faf35ba409d5705b2caadd5d796aR461-R468):
Added a test to ensure `isClientReady` is set to `false` upon component
destruction.

### Testing Enhancements:
*
[`core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/layout/withLayout.spec.ts`](diffhunk://#diff-c4b441749ccd043be6d95340a54fbc26c7466acfc6e2d4ae4672f043170f7008R1-R77):
Added comprehensive tests for the `withLayout` feature, including layout
properties and update methods.
*
[`core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/load/withLoad.spec.ts`](diffhunk://#diff-cdf3dddb12d973780196e0402f92c62736024ca9fc5b5b649f104565849d4d0eR1-R354):
Introduced tests for the `withLoad` feature, covering store
initialization, navigation, and error handling.

### Error Handling and Code Cleanup:
*
[`examples/angular/src/app/pages/services/page.service.ts`](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L30-R52):
Enhanced error handling in `fetchPage` method and improved the structure
of the `getPageAndNavigation` method to use `forkJoin` for concurrent
requests.
([examples/angular/src/app/pages/services/page.service.tsL30-R52](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L30-R52),
F6efbd08L65R84)
*
[`examples/angular/src/app/pages/pages.component.ts`](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L75-R102):
Refactored the `ngOnInit` method for better readability and added a
console log for debugging in `#setError` method.
[[1]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L75-R102)
[[2]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395R114-R116)

### Minor Code Cleanups:
*
[`examples/angular/src/app/pages/pages.component.ts`](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L1-R1):
Removed unnecessary line breaks and simplified import statements.
[[1]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L1-R1)
[[2]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L41)
*
[`examples/astro/src/components/Error.astro`](diffhunk://#diff-e67728dc39a0c654337d1f78dffb4b1edabfb1cbb0b24d4047c75bc4c4702971L3-L4):
Removed unused variables and cleaned up the code.

### Video


https://github.com/user-attachments/assets/426ef750-b367-4e05-beda-37ea44e87f66

zJaaal committed:
feat(uve): Support `editContentlet` custom action (#30456)

This pull request includes several updates to the
`DotEmaDialogComponent` and its related files, focusing on renaming the
`payload` property to `actionPayload` and adding a new `clientAction`
property. These changes aim to improve code clarity and functionality.

### Updates to `DotEmaDialogComponent`:

* Renamed the `payload` property to `actionPayload` across various
methods (`addContentlet`, `addForm`, `addWidget`, `editContentlet`,
`createContentlet`, `createContentletFromPalette`).
[[1]](diffhunk://#diff-90ca5d45502430e9bc18da8661aa443a9530d1dfe57c907854decf65176194b0L96-R112)
[[2]](diffhunk://#diff-90ca5d45502430e9bc18da8661aa443a9530d1dfe57c907854decf65176194b0L121-R141)
[[3]](diffhunk://#diff-90ca5d45502430e9bc18da8661aa443a9530d1dfe57c907854decf65176194b0L188-R190)
[[4]](diffhunk://#diff-90ca5d45502430e9bc18da8661aa443a9530d1dfe57c907854decf65176194b0L202-R209)
* Added the `clientAction` property to the component's state and related
method calls.
[[1]](diffhunk://#diff-90ca5d45502430e9bc18da8661aa443a9530d1dfe57c907854decf65176194b0L337-R340)
[[2]](diffhunk://#diff-90ca5d45502430e9bc18da8661aa443a9530d1dfe57c907854decf65176194b0L409-R414)

### Updates to `dot-ema-dialog.component.spec.ts`:

* Updated test cases to reflect the renaming of `payload` to
`actionPayload` and the addition of `clientAction`.
[[1]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4R18)
[[2]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4L181-R187)
[[3]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4L203-R210)
[[4]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4L334-R337)
[[5]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4L355-R358)
[[6]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4L382-R385)
[[7]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4L421-R427)
[[8]](diffhunk://#diff-74bc5b082cc3e4a13573517796b104be728b49d58d3cede278110b75cd7944f4L440-R445)

### Updates to `dot-ema-dialog.store.spec.ts`:

* Modified test cases to use `actionPayload` instead of `payload` and
added checks for the `clientAction` property.
[[1]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eR5)
[[2]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL47-R52)
[[3]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL60-R62)
[[4]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL69-R71)
[[5]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL85-R92)
[[6]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL117-R157)
[[7]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL148-R189)
[[8]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL162-R200)
[[9]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL171-R214)
[[10]](diffhunk://#diff-30064cf2e49d8544c8ed19f040b93774175d47edff3b01ce2b7782c3bbcd506eL190-R234)

### Additional notes:

* Added a comment in `dot-contentlet.model.ts` to highlight the
potential issues with using a flexible type for storing any kind of data
and suggested a possible refactor.
[[1]](diffhunk://#diff-6ee53bb90f1585590095ab955fc6f356346bf4ce0b4f4919e37a68489d92d48bR1-R2)
[[2]](diffhunk://#diff-6ee53bb90f1585590095ab955fc6f356346bf4ce0b4f4919e37a68489d92d48bR41)

dcolina committed:
fix(deployment): Publish Starter deployment failure due to incorrect credentials (#30482) (#30509)

### Proposed Changes
* Fixing all the environment references to 'starter'.
* A new option to include or not old assets.

### Additional Info
Related to #30482 (Publish Starter deployment failure due to incorrect
credentials).

jcastro-dotcms committed:
chore(Telemetry) #30079 : Move telemetry plugin into core - Part 1 (#30520)

### Proposed Changes
* Migrates all the code from the `Telemetry` plug-in into our `core`
project.
* This is the first part of the process, so it just takes the code "as
is" from the plugin, and places it inside the `core` project. It only
includes a few adjustments on some Javadoc, code formatting, and the
addition of the `toString()` method in a few classes.
* These changes can safely co-exist with the current `Telemetry` plugin.
* In order to have a more effective Code Review process, the goal is to
have two pull requests:
  * The first one with the existing code from the plugin -- this one.
* The second one with the code changes in `core` that effectively add
the REST Endpoint, Web Interceptor, and Quartz Job which are currently
being added via the `Activator` class.

jdotcms committed:
#30519 adding history and comments (#30527)

Adding the workflow task history and comments for edit contenlet

jgambarios committed:
Fix (CLI): Adding back builder to properly handle content type base types. (#30537)

This pull request introduces a custom builder for
`SaveContentTypeRequest` to handle the `typeInf` attribute and updates
relevant tests and methods to use this new builder. The most important
changes include adding the custom builder class, updating test cases,
and modifying methods to utilize the new builder.

### Enhancements to `SaveContentTypeRequest`:

* Added a custom `Builder` class to `AbstractSaveContentTypeRequest` to
handle the `typeInf` attribute.
(`tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/model/contenttype/AbstractSaveContentTypeRequest.java`)

### Updates to Test Cases:

* Updated imports and added new test methods in `ContentTypeAPIIT` to
test the new builder functionality.
(`tools/dotcms-cli/api-data-model/src/test/java/com/dotcms/api/ContentTypeAPIIT.java`)
[[1]](diffhunk://#diff-f2101fd155aefd1a94a63ecac9b5675e464720447b556c8b05c60c1d52648de6R24-R30)
[[2]](diffhunk://#diff-f2101fd155aefd1a94a63ecac9b5675e464720447b556c8b05c60c1d52648de6R148-R191)
* Modified existing test methods to use the new
`AbstractSaveContentTypeRequest.builder().of(contentType).build()`
pattern.
(`tools/dotcms-cli/api-data-model/src/test/java/com/dotcms/api/ContentTypeAPIIT.java`)
[[1]](diffhunk://#diff-f2101fd155aefd1a94a63ecac9b5675e464720447b556c8b05c60c1d52648de6L237-R283)
[[2]](diffhunk://#diff-f2101fd155aefd1a94a63ecac9b5675e464720447b556c8b05c60c1d52648de6L361-R407)
[[3]](diffhunk://#diff-f2101fd155aefd1a94a63ecac9b5675e464720447b556c8b05c60c1d52648de6L626-R675)
[[4]](diffhunk://#diff-f2101fd155aefd1a94a63ecac9b5675e464720447b556c8b05c60c1d52648de6L952-R1001)

### Codebase Simplification:

* Updated `ContentTypePushHandler` to use the new builder for creating
and editing content types.
(`tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/push/contenttype/ContentTypePushHandler.java`)
[[1]](diffhunk://#diff-ed28cb0bb540c2c7a442107ec9cf370696bc98f1b411311b3c2817b0c78d0968R8)
[[2]](diffhunk://#diff-ed28cb0bb540c2c7a442107ec9cf370696bc98f1b411311b3c2817b0c78d0968L81-R83)

jdotcms committed:
#30472 adding postman to check shortyId (#30514)

Adding postman to check if shortyId is there

dcolina committed:
feat: Create deployartifactjavadoc composite action base (#28550) (#30535)

### Proposed Changes
* This Pull Request introduces a new composite action named
deploy-javadoc, based on the Generate/Push Javadoc step from
maven-release-process.yml. The main objective of this action is to
modularize the behavior for generating and deploying Javadocs, providing
a more reusable and consistent solution within the CI/CD workflow.

### Key changes:
- Integration with the maven-job action to reuse artifacts from previous
stages.
- Use of Maven cache to optimize performance.
- Availability of Java classes generated during the compilation process,
essential for creating immutable objects.

### Additional Info
This PR resolves #28550 (Create deploy-javadoc composite action base).

freddyDOTCMS committed:
Issue 30468 remove calls to elastic read only command and disable es read only monitor job (#30471)

### Proposed Changes
* Remove all the Used of the ES Read Only Monitor, we still are not
removing all the logic

---------

Co-authored-by: Will Ezell <will@dotcms.com>

rjvelazco committed:
SDK: Unable to Edit Unpublished Page in Angular Example (#30542)

This pull request includes several updates to the `DotCMSPagesComponent`
in the Angular example application. The changes primarily focus on
integrating editor-related functionality and improving code formatting.

### Editor Integration:

* Added `DOTCMS_CLIENT_TOKEN` import and injected it into the
`DotCMSPagesComponent` to enable editor-related functionalities.
[[1]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L20-R26)
[[2]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395R66-R85)
* Implemented a new method `#listenToEditorChanges` to handle editor
changes and update the component state accordingly.

### Code Formatting:

* Reformatted the import statements for better readability and
maintainability.
* Adjusted the method signature formatting for `#setPageContent` to be
consistent with the rest of the codebase.

### Video


https://github.com/user-attachments/assets/7ee487ce-412f-4504-8ff1-f92dac86beee

---------

Co-authored-by: Jalinson Diaz <zjaaaldev@gmail.com>

zJaaal committed:
fix(examples): Update naming (#30540)

This pull request includes several changes to improve code consistency,
readability, and correctness across different files. The most important
changes involve import statement adjustments, formatting improvements,
and updating action names.

### Import Statement Adjustments:
*
[`examples/angular/src/app/pages/pages.component.ts`](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L7-R7):
Combined multiple imports from `@dotcms/angular` into a single line and
updated `CUSTOMER_ACTIONS` to `CLIENT_ACTIONS` from `@dotcms/client`.
[[1]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L7-R7)
[[2]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L20-R16)
*
[`examples/astro/src/react/hooks/usePageAsset.ts`](diffhunk://#diff-15921ced32cc4a42067d2c1f36400763b0d8c40193419af67ed8c8d77751c9eaL3-R3):
Updated `CUSTOMER_ACTIONS` to `CLIENT_ACTIONS` from `@dotcms/client`.
*
[`examples/nextjs/src/hooks/usePageAsset.js`](diffhunk://#diff-8fa464b336a283f1eec51ffef29c76c8ae0f2d19f32d8829851694636472b4aaL1-R4):
Updated `CUSTOMER_ACTIONS` to `CLIENT_ACTIONS` from `@dotcms/client` and
adjusted import formatting.
[[1]](diffhunk://#diff-8fa464b336a283f1eec51ffef29c76c8ae0f2d19f32d8829851694636472b4aaL1-R4)
[[2]](diffhunk://#diff-8fa464b336a283f1eec51ffef29c76c8ae0f2d19f32d8829851694636472b4aaL18-R14)
[[3]](diffhunk://#diff-8fa464b336a283f1eec51ffef29c76c8ae0f2d19f32d8829851694636472b4aaL28-R30)

### Formatting Improvements:
*
[`examples/angular/src/app/pages/pages.component.ts`](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L44-R43):
Improved formatting for better readability by aligning code and removing
unnecessary spaces.
[[1]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L44-R43)
[[2]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L58-R54)
[[3]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L68-R69)
[[4]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395R85-R120)
*
[`examples/angular/src/app/pages/services/page.service.ts`](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L22-R22):
Improved formatting and alignment for better readability.
[[1]](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L22-R22)
[[2]](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L36-R43)
[[3]](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L56-R59)
[[4]](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L74-R76)
[[5]](diffhunk://#diff-c28435c294bcf4c2159724ac9f83e80369fc594d661684535e1b1a770320b9c8L95-R88)

### Action Name Updates:
* Updated `CUSTOMER_ACTIONS.CLIENT_READY` to
`CLIENT_ACTIONS.CLIENT_READY` in various files to ensure consistency
with the action naming convention.
[[1]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395L20-R16)
[[2]](diffhunk://#diff-db19dde961c6af3dbddd86db7544e4ea4ff42baae9095cee4e3017c1d4dcb395R85-R120)
[[3]](diffhunk://#diff-15921ced32cc4a42067d2c1f36400763b0d8c40193419af67ed8c8d77751c9eaL27-R27)
[[4]](diffhunk://#diff-8fa464b336a283f1eec51ffef29c76c8ae0f2d19f32d8829851694636472b4aaL28-R30)

---------

Co-authored-by: Rafael Velazco <rjvelazco21@gmail.com>

dsolistorres committed:
#28362 : changed HostSearchOptions class to Search Type enum for clarity

dsolistorres committed:
Merge branch 'main' into issue-28362-save-site-shouldnt-deactivate-site

dsolistorres committed:
Merge branch 'main' into issue-28362-save-site-shouldnt-deactivate-site

erickgonzalez committed:
Merge branch 'main' into issue-28362-save-site-shouldnt-deactivate-site

erickgonzalez committed:
Merge branch 'main' into issue-28362-save-site-shouldnt-deactivate-site
